src/patternfly/components/NotificationBadge/notification-badge.scss

```diff
@@ -1,55 +1,108 @@
-  --pf-c-notification-badge--m-read--after--BorderColor: transparent;
-  --pf-c-notification-badge--m-unread--after--BackgroundColor: var(--pf-global--active-color--400);
+  --pf-c-notification-badge--m-read--hover--after--BorderColor: var(--pf-global--BorderColor--light-100);
 
-  position: relative;
-  display: inline-block;
-  line-height: 1;
 
-  &::after {
+  // Unread
+  --pf-c-notification-badge--m-unread--Color: var(--pf-global--Color--light-100);
+  --pf-c-notification-badge--m-unread--after--BorderColor: transparent; // remove at breaking change
+  --pf-c-notification-badge--m-unread--after--BackgroundColor: var(--pf-global--active-color--100);
+
+  // Attention
+  --pf-c-notification-badge--m-attention--Color: var(--pf-global--Color--light-100);
+  --pf-c-notification-badge--m-attention--after--BorderColor: transparent; // remove at breaking change
+  --pf-c-notification-badge--m-attention--after--BackgroundColor: var(--pf-global--danger-color--100);
+  --pf-c-notification-badge__count--MarginLeft: var(--pf-global--spacer--xs);
+
+  // Attention icon
+  --pf-c-notification-badge--pf-icon-attention-bell--LineHeight: var(--pf-global--LineHeight--sm);
+
+  position: static;
+
+  &::before { // update all --after vars to --before at breaking change
     position: absolute;
-    top: var(--pf-c-notification-badge--after--Top);
-    right: var(--pf-c-notification-badge--after--Right);
-    width: var(--pf-c-notification-badge--after--Width);
-    height: var(--pf-c-notification-badge--after--Height);
+    top: var(--pf-c-notification-badge--after--Top); // replace this var with 0 at breaking change
+    right: var(--pf-c-notification-badge--after--Right); // replace this var with 0 at breaking change
+    bottom: 0;
+    left: 0;
+    width: var(--pf-c-notification-badge--after--Width); // remove at breaking change
+    height: var(--pf-c-notification-badge--after--Height); // remove at breaking change
     content: "";
     background-color: var(--pf-c-notification-badge--after--BackgroundColor);
     border: var(--pf-c-notification-badge--after--BorderWidth) solid var(--pf-c-notification-badge--after--BorderColor);
     border-radius: var(--pf-c-notification-badge--after--BorderRadius);
     transform: translate(var(--pf-c-notification-badge--after--TranslateX), var(--pf-c-notification-badge--after--TranslateY));
   }
 
+  &:hover {
+    --pf-c-notification-badge--after--BorderColor: var(--pf-c-notification-badge--hover--after--BorderColor);
+  }
+
   & > i {
     width: var(--pf-c-notification-badge__i--Width);
     height: var(--pf-c-notification-badge__i--Height);
   }
 
-  &.pf-m-unread::after {
-    --pf-c-notification-badge--after--BackgroundColor: var(--pf-c-notification-badge--m-unread--after--BackgroundColor);
+  // set children to relative so they appear above ::before
+  & > * {
+    position: relative;
+  }
+
+  .pf-icon-attention-bell {
+    line-height: var(--pf-c-notification-badge--pf-icon-attention-bell--LineHeight);
```

Comment:
can you add `display: inline-block;` here and see if it fixes the alignment issue?